### ISSUE TITLE
Assign realm default flow not possible

### ISSUE THREAD

**Issue by timmjd** [mrparkers/terraform-provider-keycloak#298]
Within the current design it's not possible to modify the realm flows caused by a __cyclic dependency__ between __keycloak_realm__ and __keycloak_authentication_flow__:
* The creation of a flow requires a realm to be present.
* The default flow of the realm must be provided during realm creation.

```
resource "keycloak_realm" "realm" {
  realm                     = "my realm"
  browser_flow              = "my flow"
}

resource "keycloak_authentication_flow" "flow" {
  realm_id          = keycloak_realm.realm.id
  alias             = "my flow"
}
```

For a newly created realm, the presence of the __browser_flow__ value will lead to a 500 error. Patching an existing realm (by modifying the terraform and reapply) will work.

I would assume it will require an additional resource type that will take an existing realm_id and flow_id to link them.

~~The current workaround is to apply the flow to all applications of a realm separat.~~

**Comment by tomrutsaert** [comment]
You are absolutely right, and I was aware of this. The reason I did not encounter this issue initially is because in my case my realm already existed.

We would need an additional resource to solve this.

To work around the cycle dependency you can use a var which you then you use in both locations(realm.browser_flow and flow.alias), or work with a string like in your example.

The work around would be to first create the realm and flow, once everything is applied, set the flow on the realm. (with a string or variable as stated above)

**Comment by bruno-delfino1995** [comment]
@tomrutsaert I did some work around this and was planning to submit a PR with [these changes](https://github.com/mrparkers/terraform-provider-keycloak/compare/master...bruno-delfino1995:authentication-bindings). However, I didn't manage to take the time to study and create automated tests so that I could submit a proper PR.

At my company, we are using the before-mentioned code, and it worked pretty well. If you'd like to reuse that code, I have no problem with it. Or, if you prefer that I work on it and submit the PR with tests, I'd appreciate some assistance on where to start, once I have no experience neither with Go nor with Terraform Provider development.

**Comment by tomrutsaert** [comment]
Seems good! The only thing that might need to be added is to support backwards compatibility perhaps. I can imagine many people are already using the old way to set a flow.
Or perhaps we can cover this in the release notes by pointing it out. I will leave that up to @mrparkers 

I not really Golang expert nor an terraform expert either. :-) The way I learned to write these tests, is by copy pasting the test from another similar resource and adapt where needed. :-)
I do not have the time at this moment to tackle this myself, so feel free to create a PR  with some basic tests and we can work from there....

**Comment by lawrenceong** [comment]
Worked around this by using default flow on realm and base flow creation.  One done, we can run the rest via CI.  For example:

```
variable "use_default_flow" {
  type    = bool
  default = false
}
locals {
  browser_flow = var.use_default_flow ? "browser" : "my browser"
}
resource "keycloak_realm" "myrealm" {
  realm   = "myrealm"
...
  browser_flow      = local.browser_flow
...
}
resource "keycloak_authentication_flow" "my_browser" {
  realm_id    = keycloak_realm.myrealm.id
  alias       = "my browser"
  description = "browser based authentication"
}
```

With that code, initial "manual" init would be:
```
TF_VAR_use_default_flow=true terraform apply \
  -target keycloak_realm.myrealm \
  -target keycloak_authentication_flow.my_browser
```

Once done, it will be possible to apply the rest.

**Comment by BytewaveMLP** [comment]
Has there been any movement on this? It looks like the branch @bruno-delfino1995 is now gone, but I can't tell if that's because a PR was merged that contained those changes, or for other reasons. This is one of the few issues blocking WebAuthn rollout for one of my projects, and I was just curious if there's still interest in getting it working.

**Comment by msvechla** [comment]
I am interested in this as well

**Comment by mrparkers** [comment]
Hi all, none of the changed proposed above have been submitted as a PR.  If someone is willing to take the time to make these changes, I'd be happy to approve them and provide any guidance that might be necessary.  Unfortunately, I don't really have the bandwidth right now to make these changes myself.

**Comment by bruno-delfino1995** [comment]
@BytewaveMLP I've moved those changes to a repository under the company I was working at the time. They are already outdated, but maybe you can start from [them](https://github.com/mrparkers/terraform-provider-keycloak/compare/master...arquivei:authentication-bindings).

I'm not particularly fond of Go currently, and I've close to 0 knowledge in it. Therefore, I won't be submitting a PR anytime soon. Feel free to use whatever you find helpful there.

**Comment by MarS919** [comment]
I also have the need for this.

**Comment by thiagolealassis** [comment]
I am interested in this as well

**Comment by mrparkers** [comment]
FYI - The PR that adds the `keycloak_authentication_bindings` resource has been merged, and is included in the latest v3.8.0 release.
